src/main/java/de/fraunhofer/aisec/cpg/passes/CallResolver.java

```diff
@@ -278,6 +279,138 @@ private void resolveArguments(CallExpression call, RecordDeclaration curClass) {
     }
   }
 
+  /**
+   * @param callSignature Type signature of the CallExpression
+   * @param functionSignature Type signature of the FunctionDeclaration
+   * @return true if the CallExpression signature can be transformed into the FunctionDeclaration
+   *     signature by means of casting
+   */
+  private boolean compatibleSignatures(List<Type> callSignature, List<Type> functionSignature) {
+    if (callSignature.size() == functionSignature.size()) {
+      for (int i = 0; i < callSignature.size(); i++) {
+        if ((callSignature.get(i).isPrimitive() != functionSignature.get(i).isPrimitive())
+            && !TypeManager.getInstance()
+                .isSupertypeOf(functionSignature.get(i), callSignature.get(i))) {
+          return false;
+        }
+      }
+    } else {
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * Computes the implicit casts that are necessary to reach the
+   *
+   * @param call we want to find invocation targets for by performing implicit casts
+   * @param functionSignature Types of the signature of the possible invocation candidate
+   * @return List containing either null on the i-th position (if the type of i-th argument of the
+   *     call equals the type of the i-th argument of the FunctionDeclaration) or a CastExpression
+   *     on the i-th position (if the argument of the call can be casted to match the type of the
+   *     argument at the i-th position of the FunctionDeclaration). If the list is empty the
+   *     signature of the FunctionDeclaration cannot be reached through implicit casts
+   */
+  private List<CastExpression> signatureWithImplicitCastTransformation(
+      CallExpression call, List<Type> functionSignature) {
+    List<Type> callSignature = call.getSignature();
+    if (callSignature.size() == functionSignature.size()) {
+      List<CastExpression> implicitCasts = new ArrayList<>();
+
+      for (int i = 0; i < callSignature.size(); i++) {
+        Type callType = callSignature.get(i);
+        Type funcType = functionSignature.get(i);
+        if (callType.isPrimitive() && funcType.isPrimitive() && !(callType.equals(funcType))) {
+          CastExpression implicitCast = new CastExpression();
+          implicitCast.setImplicit(true);
+          implicitCast.setCastType(funcType);
+          implicitCast.setExpression(call.getArguments().get(i));
+          implicitCasts.add(implicitCast);
+        } else {
+          implicitCasts.add(null);
```

Comment:
Adding null to list can cause weird effects later. What's the point of it?

Reply:
The idea is to have the list match the signature of the function, so it is possible to match the position of the list with the position of the argument. E.g: Assuming we have the following function and call
```
void test(int a, double b, float c) {}

int main() {
test(10.0, 10.0, 3);
}
```

we would generate the following implicit cast list: 
`(Cast(implicit=true, cast-type=int), null, Cast(implicit=true, cast-type=float))`

This means that the first argument of the CallExpression requires an implicit cast to int, the second one does not need an implicit cast and the third one needs an implicit cast to float

Reply:
Might I suggest that you add this comment (maybe in shorter form) to the code snippet above? thanks!